xmtp_api_grpc/src/grpc_api_helper.rs

```diff
@@ -32,8 +32,11 @@ use xmtp_proto::{
 async fn create_tls_channel(address: String) -> Result<Channel, Error> {
     let channel = Channel::from_shared(address)
         .map_err(|e| Error::new(ErrorKind::SetupError).with(e))?
+        .initial_connection_window_size(Some((1 << 31) - 1))
         .keep_alive_while_idle(true)
-        .connect_timeout(Duration::from_secs(5))
+        .connect_timeout(Duration::from_secs(10))
+        .tcp_keepalive(Some(Duration::from_secs(15)))
+        .timeout(Duration::from_secs(120))
         .keep_alive_timeout(Duration::from_secs(25))
```

Comment:
**Purpose:** Specifies how long the client will wait for a response to a keep-alive ping before considering the connection dead.

**Functionality:** If a ping response is not received within this duration, the connection is presumed to be lost and is closed.

**Impact:** This setting is crucial for quickly detecting unresponsive connections and freeing up resources associated with them. It ensures that the client has up-to-date information on the status of connections and can react accordingly.